modules/authorization/src/admin/relations.ts

```diff
@@ -0,0 +1,141 @@
+          action: ConduitRouteActions.POST,
+          description: `Creates a new relation.`,
+          bodyParams: {
+            subject: ConduitString.Required,
+            relation: ConduitString.Required,
+            object: ConduitString.Required,
+          },
+        },
+        new ConduitRouteReturnDefinition(
+          'CreateRelation',
+          Relationship.getInstance().fields,
+        ),
+        'createRelation',
+      ),
+      constructConduitRoute(
+        {
+          path: '/relations/:id',
+          action: ConduitRouteActions.GET,
+          description: `Returns a relation.`,
+          urlParams: {
+            id: ConduitString.Required,
+          },
+        },
+        new ConduitRouteReturnDefinition(
+          'GetRelation',
+          Relationship.getInstance().fields,
+        ),
+        'getRelation',
+      ),
+      constructConduitRoute(
+        {
+          path: '/relations',
+          action: ConduitRouteActions.GET,
+          description: `Returns queried relations.`,
+          queryParams: {
+            subject: ConduitString.Optional,
+            relation: ConduitString.Optional,
+            object: ConduitString.Optional,
+            skip: ConduitNumber.Optional,
+            limit: ConduitNumber.Optional,
+            sort: ConduitString.Optional,
+          },
+        },
+        new ConduitRouteReturnDefinition('GetRelations', {
+          relations: [Relationship.getInstance().fields],
+          count: ConduitNumber.Required,
+        }),
+        'getRelations',
+      ),
+      constructConduitRoute(
+        {
+          path: '/relations/:id',
+          action: ConduitRouteActions.DELETE,
+          description: `Deletes a relation.`,
+          urlParams: {
+            id: ConduitString.Required,
+          },
+        },
+        new ConduitRouteReturnDefinition('DeleteRelation'),
+        'deleteRelation',
+      ),
+    ];
+  }
+
+  async createRelation(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+    const { subject, relation, object } = call.request.params;
+    const newRelation = await RelationsController.getInstance().createRelation(
+      subject,
+      relation,
+      object,
+    );
+    this.grpcSdk.bus?.publish(
+      'authentication:create:relation',
+      JSON.stringify(newRelation),
+    );
+    return newRelation;
+  }
+
+  async getRelation(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+    const { id } = call.request.params;
+    const found = await Relationship.getInstance().findOne({ id });
+    if (isNil(found)) {
+      throw new Error('Relation not found');
+    }
+    return found;
+  }
+
+  async getRelations(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+    const { subject, relation, resource, sort } = call.request.params;
+    const { skip } = call.request.params ?? 0;
+    const { limit } = call.request.params ?? 25;
+    const found = await Relationship.getInstance().findMany(
```

Comment:
you need to check which of the fields (subject,relation,resource) have been provided otherwise you may be looking for undefined fields